### ISSUE TITLE
[JGroups / Infinispan] Explore the ways how to automatically adjust IPv4 vs IPv6 picking algorithm on dual stack OpenShift / Kubernetes clusters

### ISSUE THREAD

**Issue by iankko** [keycloak/keycloak#19232]
### Description

Kubernetes starting from v1.20 (OpenShift from v4.8) introduced support for [dual-stack clusters](https://kubernetes.io/docs/concepts/services-networking/dual-stack/). 

Besides [other necessary configuration](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#configure-ipv4-ipv6-dual-stack), ATM there are three possible values for the configuration of the ```.spec.ipFamilyPolicy``` of a Kubernetes / OpenShift service (refer [to underlying Kubernetes documentation for further clarification](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#services) regarding meaning of each of these service settings):
* ```SingleStack```
* ```PreferDualStack```
* ```RequireDualStack```

Each of the aforementioned service ```.spec.ipFamilyPolicy``` assigns different IP family endpoints for the particular service (either just IPv4, or just IPv6 on single stack cluster, or both of them on dual-stack cluster).

Since on a dual-stack cluster, there are two IP  addresses (one from each IP family) assigned for endpoints of a service, when clustering is desired JGroups fails to by default form a JGroups cluster from underlying replicas (running Keycloak pods), because there are two types of IP addresses (two IP families, IPv4 and IPv6) that could be used for pod recognition in JGroups cluster.

In [JGroups v4.1.0](http://belaban.blogspot.com/2019/05/410-released.html) the algorithm how IPv4 vs IPv6 addresses are picked, got modified as follows:

```
  Change in how IPv4/IPv6 addresses are picked

    The new algorithm centers around bind_addr defined in the transport (UDP, TCP, TCP_NIO2); the value of this address determines how other addresses (such as loopback, site_local, global, localhost or default values) are resolved.
    Example: if bind_addr=::1, then all other addresses that are not explicitly defined will be IPv6. If bind_addr=127.0.0.1, the all other addresses will be IPv4 addresses.
    The advantage of this is that you can run a JGroups stack using IPv4 and another one using IPv6 in the same process!
    https://issues.jboss.org/browse/JGRP-2343
```

The above concludes / clarifies, the JGroups [```bind_addr```](http://www.jgroups.org/manual4/index.html#Transport) setting (the ```-Djgroups.bind_addr``` system property) is a center, around which a decision what family of IP addresses (if IPv4 or IPv6) should be assigned to particular pod.

To out-of-the-box support and simplify the JGroups (used by Infinispan) configuration on dual-stack clusters, we should explore and propose a heuristics, that would automatically adjust the IP family used for Keycloak pods, creating endpoints for the JGroups ```ping``` service, depending on the ```SingleStack```, ```PreferDualStack```, and ```RequireDualStack``` setting of the JGroups ```ping``` service and single-stack / dual-stack Kubernetes (OpenShift cluster), the Keycloak pods are running at.

**Comment by ahus1** [comment]
@pruivo - please assess this enhancement if this is within the scope of the SRE team. If you think it is, remove the other teams and assign it to the SRE team. 
There is no deadline for this task as it is an enhancement, and not a reported bug.

**Comment by pruivo** [comment]
@ahus1 I don't think it is in the SRE scope. It seems related to how the operator configures the discovery service.

**Comment by ahus1** [comment]
See #36456 for the docs change which will be published on Keycloak's website with the next upcoming 26.1 patch release. The Keycloak operator will support it out of the box with 26.2 when all goes well with #36383 

While IPv6 is still not fully supported as not all tests are completed. If new problems arise, please create a new issue.

For overall IPv6 support, follow #21277
